DataFormats/SiStripCluster/test/TestReadSiStripApproximateClusterCollection.cc

```diff
@@ -77,7 +77,7 @@ namespace edmtest {
       unsigned int j = 0;
       for (const auto& cluster : detClusters) {
         unsigned int iOffset = j + iEvent.id().event();
-        if (cluster.barycenter() != expectedIntegralValues_[1] + iOffset) {
+        if (std::round(cluster.barycenter() * 65535.0 / 770.0) != expectedIntegralValues_[1] + iOffset) {
```

Comment:
I don't think this is the way this is intended to be fixed, but rather it should be the input file to be changed according to the the [instructions](https://github.com/cms-sw/cmssw/blob/master/DataFormats/SiStripCluster/README.md)  @cms-sw/core-l2 FYI

Reply:
Right, this test should ensure that both old data and new data are read out correctly. The code here should really compare the return value of `cluster.barycenter()` to the expected value (that perhaps now should be changed to `float`?).

The change of meaning of the `barycenter_` to `compBarycenter_` should be dealt with an IO read rule in the `classes_def.xml`.